main.go

```diff
@@ -264,16 +286,53 @@ func (c AzureClient) deleteResourceGroups(ctx context.Context, numberOfResourceG
 	return nil
 }
 
-func shouldDeleteResourceGroup(input resources.Group, prefix string) bool {
+func (c AzureClient) deleteManagementGroups(ctx context.Context, prefix string, actuallyDelete bool) error {
+	var listOpts managementgroups.ListOperationOptions
+	groups, err := c.managementClient.List(ctx, listOpts)
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error obtaining Management Groups List: %+v", err)
+	}
+
+	if groups.Model == nil {
+		log.Printf("[DEBUG]   No Management Groups found")
+		return nil
+	}
+	for _, group := range *groups.Model {
+		if group.Name == nil || group.Id == nil {
+			continue
+		}
+
+		groupName := *group.Name
+		id := commonids.NewManagementGroupID(*group.Id)
+
+		if _, err := uuid.ParseUUID(groupName); err != nil {
+			log.Printf("[DEBUG]   Skipping Management Group %q", groupName)
+			continue
+		}
+		log.Printf("[DEBUG]   Deleting %s", id)
+
+		var deleteOpts managementgroups.DeleteOperationOptions
+		if _, err := c.managementClient.Delete(ctx, id, deleteOpts); err != nil {
```

Comment:
this can become:

```suggestion
		if _, err := c.managementClient.Delete(ctx, id, managementgroups.DefaultDeleteOperationOptions()); err != nil {
```